core/modules/wcontrol/SqlConnMgr.cc

```diff
@@ -53,7 +53,15 @@ void SqlConnMgr::_take(bool scanQuery) {
 void SqlConnMgr::_release() {
     --_sqlConnCount;
     --_totalCount;
-    _tCv.notify_one();
+    // All threads threads must be checked as nothing will happen if one thread is
+    // notified and it is waiting for _maxScanSqlConnections, but a different
+    // thread could use _maxSqlConnections.
+    // This shouldn't  hurt performance too much, since at any given time,
+    // very few threads should be waiting. (They can only wait when first scheduled
+    // and the scheduler is limited to about 20-30 threads.)
+    // If things are backed up, it's terribly important to run any runable
+    // threads found.
+    _tCv.notify_all();
```

Comment:
How many threads would be notified by this in the worst-case scenario? How often this synchronization call would be made? I'm just curious about the potential performance impact of this change.

Reply:
The maximum number is a result of the scheduler configuration, and that is largely a function of how many cores the worker has. This is currently about 20 on all of the systems (12 cores and the scheduler probably wont function well without at least 20 threads). If we get some systems with 30 cores, then the scheduler configuration values need to change to run more threads.

If the scheduler isn't running anything, it will wake up 20 threads. If the scheduler is busy, and all threads are running, there is nothing to wake up. If all 20 threads are waiting for an SQL connection, 20 will be woken up and 19 will go back to sleep, which is probably the worst case for us.

Reply:
Nothing will be waiting on this unless transits back to the czar are taking a long time. The SQL connections cannot be closed until all the results have been. When the worker reads all the results first, then sends them, then the worker tends to run out of memory and crash. So, the results are read and transmited in reasonable sized pieces, and the SQL connection needs to be held open until all transmits for the task are done.